rust-libretro/src/environment.rs

```diff
@@ -209,9 +255,9 @@ pub unsafe fn get_system_directory<'a>(callback: retro_environment_t) -> Option<
 pub unsafe fn set_pixel_format<F: Into<retro_pixel_format>>(
     callback: retro_environment_t,
     format: F,
-) -> bool {
```

Comment:
I suppose we could map this failure case to `EnvironmentCallError::Unsupported(_)` with a message like `the frontend did not accept <pixel format>` or similiar.
But does this provide any meaningful benefit over the general `Failure` case that is used for other environment calls that return `false` as status? :)

Reply:
Hm

> But does this provide any meaningful benefit over the general `Failure` case that is used for other environment calls that return false as status? :)

If you make the distinction, then the core can try different formats if one fails. If the environment call itself fails, then the core can just not bother.

Reply:
I’m still in favor of keeping the error as is. In case of `EnvironmentCallError::Failure` a core is encouraged to try again with a different format it supports.
But I’ll definitely go ahead and change the return type of `set_pixel_format` to return the pixel format on success, this way you can easily chain calls to the function:
```rust

let pixel_format = ctx
    .set_pixel_format(PixelFormat::XRGB8888)
    .or_else(|_| ctx.set_pixel_format(PixelFormat::RGB565))
    .or_else(|_| ctx.set_pixel_format(PixelFormat::XRGB1555))
    .expect(
        "the frondend should support one of these pixel formats: XRGB888, RGB565 or 0RGB1555",
    );
```

Note: `0RGB1555` is the default, so above code should probably we written as
```rust
let pixel_format = ctx
    .set_pixel_format(PixelFormat::XRGB8888)
    .or_else(|_| ctx.set_pixel_format(PixelFormat::RGB565))
    .unwrap_or(PixelFormat::XRGB1555);
```